beater/handlers.go

```diff
@@ -54,9 +54,24 @@ type routeMapping struct {
 
 var (
 	serverMetrics  = monitoring.Default.NewRegistry("apm-server.server")
-	requestCounter = monitoring.NewInt(serverMetrics, "requests.counter")
-	responseValid  = monitoring.NewInt(serverMetrics, "response.valid")
-	responseErrors = monitoring.NewInt(serverMetrics, "response.errors")
+	requestCounter = monitoring.NewInt(serverMetrics, "requests.count")
+	responseValid  = monitoring.NewInt(serverMetrics, "response.valid.count")
+	responseErrors = monitoring.NewInt(serverMetrics, "response.errors.count")
+
+	statusQueueFull        = 5030
+	statusConcurrencyLimit = 5031
+	statusChannelClosed    = 5032
+	errMap                 = map[int]*monitoring.Int{
+		http.StatusBadRequest:       monitoring.NewInt(serverMetrics, "response.errors.badrequest"),
+		http.StatusForbidden:        monitoring.NewInt(serverMetrics, "response.errors.forbidden"),
+		http.StatusUnauthorized:     monitoring.NewInt(serverMetrics, "response.errors.unauthorized"),
+		http.StatusTooManyRequests:  monitoring.NewInt(serverMetrics, "response.errors.ratelimit"),
+		http.StatusMethodNotAllowed: monitoring.NewInt(serverMetrics, "response.errors.method"),
+		statusQueueFull:             monitoring.NewInt(serverMetrics, "response.errors.queue"),
+		statusConcurrencyLimit:      monitoring.NewInt(serverMetrics, "response.errors.concurrency"),
+		statusChannelClosed:         monitoring.NewInt(serverMetrics, "response.errors.closed"),
+	}
+	errCountMissing = "metrics.apm-server.server.response.errors count missing for"
```

Comment:
I think this assumes `response_code` will come right after in the error message, but the `response_code` will be just another key/value in the log output, coming out in random order.

Something like this:
```
metrics.apm-server.server.response.errors count missing for      {"key1": "value1", "response_code": "999"}
```
so maybe this should be something like:

`internal counter missing for given response code`

or similar